pkg/types/gcp/machinepools.go

```diff
@@ -18,6 +18,11 @@ type MachinePool struct {
 	// +optional
 	OSDisk `json:"osDisk"`
 
+	// OSImage defines a custom image for instance.
+	//
+	// +optional
+	OSImage `json:"osImage"`
```

Comment:
When we add new optional fields in the installconfig we need to make them pointers with omitempty. Hive vendors the installer and uses it to produce serialized install configs, which need to be backwards compatible. Go will only omit the struct from the serialization if it's a null pointer. Also, there's no need to embed this, so:

```go
OSImage *OSImage `json:"osImage,omitempty"`
```